packages/node/src/behavior/system/commissioning/CommissioningClient.ts

```diff
@@ -49,38 +59,42 @@ export class CommissioningClient extends Behavior {
         }
 
         this.reactTo((this.endpoint as Node).lifecycle.partsReady, this.#initializeNode);
+        this.reactTo(this.events.peerAddress$Changed, this.#peerAddressChanged);
     }
 
     commission(passcode: number): Promise<ClientNode>;
 
     commission(options: CommissioningClient.CommissioningOptions): Promise<ClientNode>;
 
-    async commission(options: number | CommissioningClient.CommissioningOptions) {
-        if (typeof options !== "object") {
+    async commission(options: number | string | CommissioningClient.CommissioningOptions) {
+        // Commissioning can only happen once
+        const node = this.endpoint as ClientNode;
+        if (this.state.peerAddress !== undefined) {
+            throw new ImplementationError(`${node} is already commissioned`);
+        }
+
+        if (typeof options === "number") {
```

Comment:
Hm ... in fact both are numbers ... you can determine by length? passcode max is 99999999. Pairingcode is minimum 11 digits